File: Day-04/solution.js
import readline from 'readline';

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question('Enter a sentence: ', (sentence) => {
    if (sentence.trim().length === 0) {
        console.log("Empty sentence!");
        rl.close();

        return;
    }

    const words = sentence.split(' ').filter(Boolean);

    const longestWord = words.reduce(
        (carry, word) => {
            word = word.replace(/^[.,!?;:(){}\[\]*]+|[.,!?;:(){}\[\]*]+$/g, '');

            return word.length > carry.length ? word : carry;
        },
        ''
    );

    console.log(`The longest word is: ${longestWord}`);

    rl.close();
});

File: Day-04/solution.py
from functools import reduce

sentence = input('Enter a sentence: ')

if not sentence.strip():
    print("Empty sentence!")
    exit()

words = sentence.split()


def find_longest_word(acc, word):
    word = word.strip(".,!?;:(){}[]*")

    return word if len(word) > len(acc) else acc


longest_word = reduce(find_longest_word, words, "")

print(f"The longest word is: {longest_word}")
